src/runtime/pythonengine.cs

```diff
@@ -406,15 +393,47 @@ public static PyObject ModuleFromString(string name, string code)
 
 
         /// <summary>
+        /// Eval Method
+        /// </summary>
+        /// <remarks>
+        /// Evaluate a Python expression and returns the result.
+        /// It's a subset of Python eval function.
+        /// </remarks>
+        public static PyObject Eval(string code, IntPtr? globals = null, IntPtr? locals = null)
+        {
+            PyObject result = RunString(code, globals, locals, RunFlagType.Eval);
+            return result;
+        }
+
+
+        /// <summary>
+        /// Exec Method
+        /// </summary>
+        /// <remarks>
+        /// Run a string containing Python code. 
+        /// It's a subset of Python exec function.
+        /// </remarks>
+        public static void Exec(string code, IntPtr? globals = null, IntPtr? locals = null)
+        {
+            PyObject result = RunString(code, globals, locals, RunFlagType.File);
+            if (result.obj != Runtime.PyNone)
```

Comment:
@vmuriart RunString will always return None when passing the last parameter Runtime.Py_file_input. So this check failed only when there is a runtime bug of CPython. It can be removed.